demonstrations/tutorial_qnn_multivariate_regression.py

```diff
@@ -0,0 +1,253 @@
+we want to fit
+3. Obtain a partial Fourier series for the target function; since the function is not periodic, this partial
+Fourier series will only approximate the function in the region we will use for training.
+4. Plot the optimized circuit expectation value against the exact function to compare the two.
+What is a quantum model?
+------------
+A quantum model :math:`g_{\vec{\theta}}(\vec{x})` is the expectation value of some observable : math : `M` with
+respect to a state prepared by a parameterized circuit :math:`U(\vec{x}, \vec{\theta})`:
+.. math:: g_{\vec{\theta}}(\vec{x}) = \langle 0 | U^\dagger (\vec{x}, \vec{\theta}) M U(\vec{x}, \vec{\theta}) | 0 \rangle.
+By repeatedly running the circuit with a set of parameters :math:`\vec{\theta}` and set of data points :math:`\vec{x}`, we can
+approximately find the expectation value with respect to the observable :math:`M`. Then, the expectation value can be
+optimized with respect to some loss function by adjusting :math:`\vec{\theta}`.
+What are we using the variational circuit for?
+------------
+In this example, we will use a variational quantum circuit to find the Fourier series that
+approximates the function :math:`f(x_1, x_2) = \frac{1}{2} \left( x_1^2 + x_2^2 \right)`. The variational circuit that we are using is made up of :math:`L` layers. Each layer consists of a *data encoding block*
+:math:`S(\vec{x})` and a *training block* :math:`W(\vec{\theta})`. The overall circuit is:
+.. math:: U(x, \vec{\theta}) = W^{(L+1)}(\vec{\theta}) S(\vec{x}) W^{(L)} (\vec{\theta}) \ldots W^{(2)}(\vec{\theta}) S(\vec{x}) W^{(1)}(\vec{\theta}).
+The training blocks :math:`W(\vec{\theta})` depend on a vector of parameters :math:`\vec{\theta}` that can be optimized classically.
+.. figure:: ../_static/demonstration_assets/qnn_multivariate_regression/qnn_circuit.png
+    :align: center
+    :width: 90%
+We will build a circuit such that the expectation value of the :math:`Z` observable is a partial Fourier series
+that approximates :math:`f(\vec{x})`, i.e.,
+.. math:: f(\vec{x}) \approx \sum_{\vec{\omega} \in \Omega} c_\vec{\omega} e^{i \vec{\omega} \vec{x}} = g_{\vec{\theta}}.
+Then, we can directly plot the partial Fourier series. We can also apply a Fourier transform to
+:math:`g_{\vec{\theta}}`, so we can obtain the Fourier coefficients :math:`c_\vec{\omega}`. To know more about how to obtain the 
+Fourier series check out these two related tutorials [#demoschuld]_,[#qibodemo]_.
+How do we actually construct the quantum circuit?
+------------
+"""
+##############################################################################
+# First, let's import the necessary libraries and seed the random number generator.
+# We will also define the device, which has two qubits, using qml.device.
+import matplotlib.pyplot as plt
+import pennylane as qml
+from pennylane import numpy as np
+import jax
+from jax import numpy as jnp
```

Comment:
Would be nice to briefly mention below or above this code snippet what is `jax` and why we use it, and maybe link to documentation or introductory tutorial on what `jax` does. Or import it later at the optimization step where you explain there.